### PR TITLE
AI Gear Macro

### PR DESCRIPTION
**Pull request by Freddo3000**
Adds a simple module that bulk applies AI gear. Unit type is decided via `BIS_fnc_objectType`

3DEN module
---
image
Zeus module
---
image
Role configuration
---
Used roles and their probabilities can be defined `class TMF_AIGear`, which can be defined in `missionConfigFile` as well as `configFile`.
```hpp
class TMF_AIGear {
    // Code used to determine unit type, 
    // must return string equal to one of the arrays below.
    code = "([_this] call BIS_fnc_objectType) select 1";

    // Type arrays, follows selectRandomWeighted syntax
    MG[] = {
        "ar",3,
        "mmgg",1
    };
    AT[] = {
        "rat",3,
        "matg",1
    };
    Officer[] = {
        "pl",1
    };
    Pilot[] = {
        "pc",3,
        "jp",1
    };
    Medic[] = {
        "m",1
    };
    Sniper[] = {
        "sn",1,
        "dm",1
    };

    // Default if determined type isn't one of the above arrays
    default[] = {
        "r", 10,
        "ftl", 2,
        "aar", 2,
        "eng",1,
        "sl",1
    };
};
```

Todo
---
- [x] Add Zeus UI
- [ ] Test in MP
- [x] Add config options for loadout distribution
- [x] Tweak loadout handling logic
- [ ] Add wiki entry

### CODE REVIEW

**Comment by beaar** [comment]
I think especially with a Zeus module this can be super useful 👍

Module Info should mention what loadout roles are used

**Comment by Freddo3000** [comment]
Ready for review

File: addons/assigngear/CfgScriptPaths.hpp [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class CfgScriptPaths {
+    ADDON = "x\tmf\addons\assigngear\UI\";
+}
```
Comment (Freddo3000) [inline addons/assigngear/CfgScriptPaths.hpp]: missing semicolon

File: addons/assigngear/UI/TMF_AttributesFaction.sqf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            _ctrlCombo lbSetData [_id,_data];
+            _ctrlCombo lbSetPictureRight [_id,_icon];
+        } forEach _factionClasses;
+        lbSort _ctrlCombo;
+        _ctrlCombo lbSetCurSel 0;
+    };
+    #ifdef CURATOR_ATTRIBUTE
+    case "onUnload": {};
+    case "confirmed": {
+        _unit setVariable ["Faction",_ctrlCombo lbData lbCurSel _ctrlCombo];
```
Comment (Sniperhid) [inline addons/assigngear/UI/TMF_AttributesFaction.sqf]: This is a nitpick but generically named variables like "Faction" and "Updated" on a unit they could collide with other mods/scripts so using `GVAR` prefixes is advisable. As you pointed out to me this is a module. I'd suggest also renaming the `_unit` variable to be either `_module` or `_logic` which makes it easier for the reader. I tend to assume a variable named `_unit` would be a man object.

As it is a module other mods shouldn't be setting variables on it, so I'll settle for renaming the variable to `_logic` or `_module` (or something similiar)

**Review by Sniperhid — CHANGES_REQUESTED** [review]
Things generally look good from a good point of view, but minor nitpick

File: addons/assigngear/UI/TMF_AttributesFaction.sqf (hunk 2 above)
Comment (Freddo3000) [reply addons/assigngear/UI/TMF_AttributesFaction.sqf]: The `updated` part is simply following the BI curator module standard, which uses
```sqf
waituntil {!isnil {_logic getvariable "updated"} || isnull _logic};
```

**Review by Sniperhid — APPROVED** [review]

**Comment by Sniperhid** [comment]
Any feedback @headswe ?

**Comment by headswe** [comment]
> Any feedback @headswe ?

It looks fine to me.

**Pull request merged**